core/src/components/input/test/basic/input.e2e.ts

```diff
@@ -201,3 +201,74 @@ test.describe('input: clear button', () => {
     await expect(nativeInput).toBeFocused();
   });
 });
+
+test.describe('input: clearOnEdit', () => {
+  test('should clear the input on first keystroke of input being focused', async ({ page }) => {
+    await page.setContent(`<ion-input value="some value" clear-on-edit="true"></ion-input>`);
+
+    const input = page.locator('ion-input');
+
+    await input.click();
+    await input.type('h');
+
+    expect(await input.evaluate((el: HTMLIonInputElement) => el.value)).toBe('h');
+
+    await input.type('ello world');
+
+    expect(await input.evaluate((el: HTMLIonInputElement) => el.value)).toBe('hello world');
+  });
+
+  test('should not clear the input when pressing Enter', async ({ page }) => {
+    await page.setContent(`<ion-input value="some value" clear-on-edit="true"></ion-input>`);
+
+    const input = page.locator('ion-input');
+
+    await input.click();
+    await page.keyboard.press('Enter');
+
+    expect(await input.evaluate((el: HTMLIonInputElement) => el.value)).toBe('some value');
+  });
+
+  test('should not clear the input if it does not have an initial value when typing', async ({ page }) => {
+    await page.setContent(`<ion-input value="" clear-on-edit="true"></ion-input>`);
+
+    const input = page.locator('ion-input');
+
+    await input.click();
+    await input.type('hello world');
+
+    expect(await input.evaluate((el: HTMLIonInputElement) => el.value)).toBe('hello world');
+  });
+
+  test('should emit ionChange once on blur', async ({ page }, testInfo) => {
```

Comment:
I feel like this test should be in the `input-events.e2e.ts` file instead, since that's where the `ionInput` tests for `clearOnEdit` are located.

Reply:
I opted for locating this with the existing `clearOnEdit` tests, since the bug and behavior was specific to that feature being enabled.